packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/MediaLibraryModule.kt

```diff
@@ -366,6 +385,26 @@ class MediaLibraryModule : Module() {
     fun runWithPermissions(permissionsWereGranted: Boolean)
   }
 
+  private fun hasReadPermissions(): Boolean {
+    val permissions = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+      arrayOf(READ_MEDIA_IMAGES, READ_MEDIA_AUDIO, READ_MEDIA_VIDEO)
+    } else {
+      arrayOf(READ_EXTERNAL_STORAGE, WRITE_EXTERNAL_STORAGE)
+    }
+
+    return appContext.permissions
+      ?.hasGrantedPermissions(*permissions)
+      ?.not() ?: false
+  }
+
+  private fun hasWritePermissions() = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+    false
```

Comment:
Is this an oversight? Does this not lead to not being able to save anything at all anymore?

Reply:
I see, the problem I spot then is that `saveToLibraryAsync()` is wrapped in `throwUnlessPermissionsGranted`, therefore this seems to throw unconditionally even if the necessary permissions are actually granted.

Reply:
Will make an issue if you confirm that I did not oversee anything!

Reply:
👌 the check might be a bit confusing. It will throw if `isMissingWritePermissions` returns true but that can't happen on android 13.